packages/core/src/components/chat.ts

```diff
@@ -19,34 +19,42 @@ export interface ReceivedChatMessage extends ChatMessage {
 export type MessageEncoder = (message: ChatMessage) => Uint8Array;
 /** @public */
 export type MessageDecoder = (message: Uint8Array) => ReceivedChatMessage;
+/** @public */
+export type ChatOptions = {
+  messageEncoder?: (message: ChatMessage) => Uint8Array;
+  messageDecoder?: (message: Uint8Array) => ReceivedChatMessage;
+  channelTopic?: string;
+};
+
+type RawMessage = {
+  payload: Uint8Array;
+  topic: string | undefined;
+  from: Participant | undefined;
+};
 
 const encoder = new TextEncoder();
 const decoder = new TextDecoder();
 
+const topicSubjectMap: Map<string, Subject<RawMessage>> = new Map();
```

Comment:
@Ocupe currently, this gets never cleared/disposed. We won't want to merge it as is. 
I was thinking about when the best way to dispose of specific entries would be. 
My thoughts:
- do some ref counting and dispose when refCount === 0
- only dispose on room disconnect (we'd have to be sure that we catch that event)